json-schema/json-schema-generic/src/main/scala/endpoints/generic/JsonSchemas.scala

```diff
@@ -103,9 +104,9 @@ trait JsonSchemas extends algebra.JsonSchemas {
       taggedTail: GenericTagged[T]
     ): GenericTagged[FieldType[L, H] :+: T] =
       new GenericTagged[FieldType[L, H] :+: T] {
-        def jsonSchema: Tagged[FieldType[L, H] :+: T] = {
-          val taggedHead = recordHead.jsonSchema.tagged(labelHead.value.name)
-          taggedHead.orElse(taggedTail.jsonSchema).invmap[FieldType[L, H] :+: T] {
+        def jsonSchema(docs: List[Option[documentation]]): Tagged[FieldType[L, H] :+: T] = {
+          val taggedHead = recordHead.jsonSchema(Nil).tagged(labelHead.value.name)
```

Comment:
This `Nil` makes me a bit sad :(

Reply:
Why don’t we use `docs.head`? (I’m not familiar with `shapeless.Annotation`)

Reply:
Shortest reason:
`docs.head` returns `Option[documentation]`, not `List[Option[documentation]]`.

Longer reason:
I haven't established a strategy to document each coproduct branch yet. In this case, the result of `shapeless.Annotations[documentation, A]` is the list of `@documentation` annotating each branch of `sealed trait A` like this:

```scala
sealed trait Foo
@documentation("barDoc") case object Bar extends Foo
case object Baz extends Foo

assert(Annotations[documentation, A].apply == List(Some(documenation("barDoc")), None))
```

Since `recordHead` is already `GenericRecord`, I'd rather expand `@documentation` to annotate the class itself and let `recordGeneric` handle it, which is not implemented in this PR yet.

Reply:
Related discussion: https://github.com/milessabin/shapeless/issues/866#issuecomment-426913817